dotnet-authserver/src/TeacherIdentity.AuthServer/Oidc/TeacherIdentityApplicationManager.cs

```diff
@@ -36,6 +36,43 @@ public override async ValueTask PopulateAsync(OpenIddictApplicationDescriptor de
         }
     }
 
+    public async ValueTask<bool> ValidateRedirectUriDomain(Application application, string address, CancellationToken cancellationToken = default)
+    {
+        ArgumentNullException.ThrowIfNull(application);
+        ArgumentException.ThrowIfNullOrEmpty(address);
+
+        if (!Uri.TryCreate(address, UriKind.Absolute, out var addressUri))
+        {
+            return false;
+        }
+
+        var addressAuthority = addressUri.GetLeftPart(UriPartial.Authority);
+
+        foreach (var uri in await Store.GetRedirectUrisAsync(application, cancellationToken))
+        {
+            var authority = new Uri(uri).GetLeftPart(UriPartial.Authority);
+
+            if (WildcardPathSegmentPattern().IsMatch(authority))
+            {
+                var pattern = $"^{Regex.Escape(authority).Replace("__", ".*")}$";
+
+                if (Regex.IsMatch(addressAuthority, pattern))
+                {
+                    return true;
+                }
+            }
+            else
+            {
+                if (authority.Equals(addressAuthority))
```

Comment:
Would it be better to check for equality before trying to find a match with regex?

Reply:
Probably can drop the else statement as well

Reply:
Initially it was for symmetry with the existing `ValidateRedirectUriAsync` method (which was written to be as close as the base implementation as possible). I've flipped to check the direct match first.